src/content/get-started/async.md

```diff
@@ -0,0 +1,374 @@
+    Now, use the `http` client to make an HTTP `GET` request to the URL
+    you just constructed. The `await` keyword pauses the execution of
+    `getWikipediaArticle` until the `client.get(url)` call completes and
+    returns an `http.Response` object.
+
+    After the request completes, check the `response.statusCode` to
+    ensure the request was successful (a status code of `200` means OK).
+    If successful, return the `response.body`, which contains the
+    fetched data (in this case, raw JSON). If the request failed, return
+    an informative error message.
+
+    Add these lines after the `Uri` construction within `getWikipediaArticle`:
+
+    ```dart
+    Future<String> getWikipediaArticle(String articleTitle) async {
+      final http.Client client = http.Client();
+      final Uri url = Uri.https(
+        'en.wikipedia.org',
+        '/api/rest_v1/page/summary/$articleTitle',
+      );
+      final http.Response response = await client.get(url); // Make the HTTP request
+
+      if (response.statusCode == 200) {
+        return response.body; // Return the response body if successful
+      }
+
+      // Return an error message if the request failed
+      return 'Error: Failed to fetch article "$articleTitle". Status code: ${response.statusCode}';
+    }
+    ```
+
+### Task 4: Refactor `searchWikipedia` into `runApp` and integrate the API call
+
+In Chapter 2, you added a `searchWikipedia` function. To make the code more organized and align with common Dart application patterns, you'll now rename that function to `runApp` and integrate the API call into it. This `runApp` function will house the core logic for handling the `wikipedia` command.
+
+1.  **Rename `searchWikipedia` to `runApp` and update its signature:**
+    Locate your `searchWikipedia` function and rename it to `runApp`. Also, update its signature to be `async` as it will now perform asynchronous operations.
+
+    Your `runApp` function should now look like this (initial part):
+
+    ```dart
+    // ... (your existing main function)
+
+    void runApp(List<String>? arguments) async { // Renamed from searchWikipedia and added 'async'
+      late String? articleTitle;
+
+      // If the user didn't pass in arguments, request an article title.
+      if (arguments == null || arguments.isEmpty) {
+        print('Please provide an article title.');
+        articleTitle = stdin.readLineSync(); // Await input from the user
+        // We'll add error handling for null input here in a moment
+      } else {
+        // Otherwise, join the arguments into the CLI into a single string
+        articleTitle = arguments.join(' ');
+      }
+
+      // ... rest of the function
+    }
+
+    // ... (your existing printUsage() function)
+    ```
+    * `void runApp(List<String>? arguments) async`: Notice that the function is now `async`. This is essential because it will call `getWikipediaArticle`, which is an `async` function itself and we'll need to `await` its result.
```

Comment:
Format this to look like the other sections that have a bulleted list below a code snippet.

Reply:
I added the "list title" - "Highlights from the preceding code:"